### PR TITLE
[Enhancement] add more error info to s3 sdk

### PR DESCRIPTION
**Pull request by dirtysalt**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

Try to expose more error info of accessing s3. The possible used field are

1. [x] http code (403)
2. [x] message: no response body
3. exception: null
4. response headers:
    1. connection / keepalive
    2. content-length: 324
    3. content-type: application/xml
    4. date: Sat, 06 Aug 2022 04:46:56 GMT
    5. server: AliyunOSS
    6. x-amz-request-id: 62EDF240825092313227F4B8
    7. x-oss-server-time: 22
5. [x] errortype: Aws::S3::S3Errors::ACCESS_DENIED

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

File: be/src/io/s3_input_stream.cpp [hunk 1]
```diff
@@ -30,7 +30,9 @@ StatusOr<int64_t> S3InputStream::read(void* out, int64_t count) {
         _offset += body.gcount();
         return body.gcount();
     } else {
-        return Status::IOError(outcome.GetError().GetMessage());
+        const auto& error = outcome.GetError();
+        return Status::IOError(fmt::format("code={}(SdkErrorType:{}), message={}", error.GetResponseCode(),
```
Comment (DorianZheng) [inline be/src/io/s3_input_stream.cpp:34]: extract as a function. us can be used in multiple place

Comment (dirtysalt) [reply be/src/io/s3_input_stream.cpp]: OK. Fixed.

**Review by DorianZheng — APPROVED** [review]
LGTM, Plz cp to branch-2.3 and branch-2.4

**Comment by dirtysalt** [comment]
run all

**Review by gengjun-git — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by stdpain — APPROVED** [review]

**Comment by dirtysalt** [comment]
@mergifyIO backport branch-2.3

**Comment by dirtysalt** [comment]
@MergifyIO backport branch-2.4

**Pull request merged**